### ISSUE TITLE
Tests take too long to be discovered and run

### ISSUE THREAD

**Issue by osotorrio** [xunit/xunit#1376]
**Preconditions**

1. I am using Visual Studio Community 2015 Update 3.
2. NUnit 3 Test Adapter extension is installed in VS.
3. The test project targets .net Framework 4.6.1
4. I have deleted the folder %Temp%/VisualStudioTestExplorerExtensions

5. This is my packages.config file:

```xml  
  <package id="xunit" version="2.2.0" targetFramework="net461" />
  <package id="xunit.abstractions" version="2.0.1" targetFramework="net461" />
  <package id="xunit.assert" version="2.2.0" targetFramework="net461" />
  <package id="xunit.core" version="2.2.0" targetFramework="net461" />
  <package id="xunit.extensibility.core" version="2.2.0" targetFramework="net461" />
  <package id="xunit.extensibility.execution" version="2.2.0" targetFramework="net461" />
  <package id="xunit.runner.console" version="2.2.0" targetFramework="net461" developmentDependency="true" />
  <package id="xunit.runner.visualstudio" version="2.2.0" targetFramework="net461" developmentDependency="true" />
```

**Steps**

I am following the steps explained here: [https://xunit.github.io/docs/getting-started-desktop.html](https://xunit.github.io/docs/getting-started-desktop.html)

**The issue**

When running the test with the console runner:

1. It takes long time to start running. Around 1 minute.
2. Once it starts running, it takes 10 seconds to run only 2 test.

When running the test with VS runner:

1. First, it tries to run the test with NUnit 3 addapter.
2. Then, XUnit takes long time to discover the test. Around 1 minute.
3. Then, XUnit takes long time to run the test. Around 2 minutes.

**Output**

------ Run test started ------
NUnit Adapter 3.8.0.0: Test execution started
Running all tests in C:\Projects\Temporal\MyFirstUnitTest\MyFirstUnitTest\bin\Debug\MyFirstUnitTest.dll
NUnit couldn't find any tests in C:\Projects\Temporal\MyFirstUnitTest\MyFirstUnitTest\bin\Debug\MyFirstUnitTest.dll
NUnit Adapter 3.8.0.0: Test execution complete
[xUnit.net 00:01:00.5250702]   Discovering: MyFirstUnitTest
[xUnit.net 00:01:10.9855415]   Discovered:  MyFirstUnitTest
[xUnit.net 00:02:11.6721519]   Starting:    MyFirstUnitTest
[xUnit.net 00:02:32.3150582]     MyFirstUnitTest.Class1.FailingTest [FAIL]
[xUnit.net 00:02:32.3236083]       Assert.Equal() Failure
[xUnit.net 00:02:32.3246422]       Expected: 5
[xUnit.net 00:02:32.3252201]       Actual:   4
[xUnit.net 00:02:32.3272116]       Stack Trace:
[xUnit.net 00:02:32.3282522]         C:\Projects\Temporal\MyFirstUnitTest\MyFirstUnitTest\Class1.cs(21,0): at MyFirstUnitTest.Class1.FailingTest()
[xUnit.net 00:02:32.3424911]   Finished:    MyFirstUnitTest
========== Run test finished: 2 run (**0:02:33.877811**) ==========

**Comment by bradwilson** [comment]
Can you compare this with times using the console runner?

I find two things very suspicious here. One, regardless of why it takes 10s to discover and 21s to run the tests, Visual Studio has somehow made the problem worse by taking 1 minute to take the discovered tests and run them. This feels like an under-resourced machine (not enough RAM, very slow disk, etc.). Can you reproduce this on any other machine? Can you provide the specifications for the machine in question?

**Comment by osotorrio** [comment]
The machine specifications are as follow:

Processor: Intel(R) Core(TM) i7-4510U CPU @ 2.00GHz 2.60GHz
RAM: 8 GB
ROM: 250 SSD
Sytem type: x64-based processor
OS: Windows 10 Home

What I found suspicious is the following part of the previous output:

> NUnit Adapter 3.8.0.0: Test execution started
> Running all tests in C:\Projects\Temporal\MyFirstUnitTest\MyFirstUnitTest\bin\Debug\MyFirstUnitTest.dll
> **NUnit couldn't find any tests** in C:\Projects\Temporal\MyFirstUnitTest\MyFirstUnitTest\bin\Debug\MyFirstUnitTest.dll
> NUnit Adapter 3.8.0.0: Test execution complete

Why NUnit Adapter should even try to run the tests before the XUnit runner?

Therefore, I uninstalled NUnit Adapter 3.8.0.0 from my VS. Then, I restarted my machine (thinking maybe in some kind of residual cache). And I installed back NUnit Adapter. 

Now everything is working smoothly.

------ Run test started ------
[xUnit.net 00:00:00.3080979]   Discovering: MyFirstUnitTest
[xUnit.net 00:00:00.4237529]   Discovered:  MyFirstUnitTest
[xUnit.net 00:00:00.8038637]   Starting:    MyFirstUnitTest
[xUnit.net 00:00:01.0986263]   Finished:    MyFirstUnitTest
========== Run test finished: 2 run (0:00:01.2563356) ==========
